### PR TITLE
[onert] Add shape inference for reverse op.

### PR DESCRIPTION
**Pull request by Hyunjun85**
Add shape inference for reverse op..

Issue: https://github.com/Samsung/ONE/issues/2103

ONE-DCO-1.0-Signed-off-by: H Kim <hyunjun2.kim@samsung.com>

### CODE REVIEW

**Comment by Hyunjun85** [comment]
> just like this PR, if input&output shape is the same,
> then can we skip adding TC for those cases?

About this op, the input and output shape always same. So it can be covered with existing tc.

**Comment by Hyunjun85** [comment]
Rebase from master code and add test cases for this.

File: runtime/onert/core/src/util/shapeinf/Reverse.cc [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#include "util/ShapeInference.h"
+
+namespace onert
+{
+namespace shape_inference
+{
+
+void StaticInferer::visit(const ir::operation::Reverse &op)
+{
+  const auto input_idx{op.getInputs().at(ir::operation::Reverse::Input::INPUT)};
```
Comment (YiHyunJin) [inline runtime/onert/core/src/util/shapeinf/Reverse.cc:26]: Why not use handleSimpleUnaryOp?

Comment (Hyunjun85) [reply runtime/onert/core/src/util/shapeinf/Reverse.cc]: I miss that.

Comment (Hyunjun85) [reply runtime/onert/core/src/util/shapeinf/Reverse.cc]: Applied.

**Comment by Hyunjun85** [comment]
Rebase master and applied review.

**Pull request merged**